src/test/java/org/fogbowcloud/saps/engine/core/archiver/storage/SwiftPermanentStorageTest.java

```diff
@@ -299,7 +299,7 @@ public void testToDeleteFailureTaskWithoutArchiverInDebugMode() throws Exception
             return null;
         }).when(swiftAPIClient).deleteFile(Mockito.eq(MOCK_CONTAINER_NAME), Mockito.argThat(new StartsWith(MOCK_SWIFT_FOLDER_PREFIX + "/" + Dirs.Task02)));
 
-        boolean deleteTask02 = permanentStorage.archive(task02);
+        boolean deleteTask02 = permanentStorage.delete(task02);
```

Comment:
a delete method that returns a boolean might be a bad smell. Shouldn't it throw and exception if something goes wrong?

Reply:
there is not an issue for this pr tough, once the delete method is not modified here..

Reply:
> a delete method that returns a boolean might be a bad smell. Shouldn't it throw and exception if something goes wrong?

not always. returning a boolean is ok, in some cases. The key point is to consider what are we going to do with the boolean/exception. if we are going to catch the exception and to branch the logic in the upper level it would be better to use a boolean.

Reply:
I kept the way it was being made and didn't add any fixme about it. Do you think it is better to fix a problem with this method?

Reply:
I had already added a Fixme about it on Permanent Storage Interface.